enigma-core/enclave/src/lib.rs

```diff
@@ -279,9 +279,14 @@ unsafe fn store_delta_and_state(db_ptr: *const RawPointer, delta: &Option<Encryp
         Some(d) => {
             let enc_state = km_t::encrypt_state(state.clone())?;
             enigma_runtime_t::ocalls_t::save_delta(db_ptr, d)?;
-            // todo: in case the delta was stored and the state wasn't, we should revert the delta- create a new ocall
-            enigma_runtime_t::ocalls_t::save_state(db_ptr, &enc_state)?;
-            Ok(())
+            match enigma_runtime_t::ocalls_t::save_state(db_ptr, &enc_state) {
```

Comment:
I think that it is better to do: `if let Err(e) = enigma_runtime_t::ocalls_t::save_state(db_ptr, &enc_state) {...}` or use `is_err` than `match` here.

Reply:
https://github.com/enigmampc/enigma-core/pull/224/commits/cfee01897053a154c5ba21683776d254758143f0